gslab_scons/tests/test_release_function.py

```diff
@@ -13,25 +17,57 @@
+# specified as arguments in test methods in the opposite order as the
+# patches are listed below. 
+#
+# Note that time.sleep is just mocked to do nothing.
+path = 'gslab_scons._release_tools'
+patch_collection = \
+    lambda f: mock.patch('%s.time.sleep'    % path, mock.MagicMock())(\
+              mock.patch('%s.os.path.isdir' % path)(\
+              mock.patch('%s.shutil.rmtree' % path)(                  \
+              mock.patch('%s.upload_asset'  % path)(                  \
+              mock.patch('%s.requests.session' % path)(               \
+              mock.patch('%s.getpass.getpass' % path)(
+              mock.patch('%s.os.makedirs' % path)(
+              mock.patch('%s.shutil.copy' % path)(
+              mock.patch('%s.shutil.make_archive' % path)(
+              mock.patch('%s.shutil.move' % path)(\
+                f \
+              ))))))))))
+
+# Standard release() arguments
+standard_args = {'vers':              'test_version',
+                 'DriveReleaseFiles': ['paper.pdf', 'plot.pdf'],
+                 'local_release':     'root/release/repo-test_branch',
+                 'org':               'org',
+                 'repo':              'repo',
+                 'target_commitish':  'test_branch',
+                 'zip_release':       True}
+
+# This is mocked-up output from requests.session.get(path) based on 
+# members of standard_args
+org_repo = (standard_args['org'], standard_args['repo'])
+#====================================================================
 
 class TestReleaseFunction(unittest.TestCase):
 
-    @mock.patch('gslab_scons._release_tools.upload_asset')
-    @mock.patch('gslab_scons._release_tools.requests.session')
-    @mock.patch('gslab_scons._release_tools.getpass.getpass')
-    @mock.patch('gslab_scons._release_tools.time.sleep') # For skipping time delay
-    @mock.patch('gslab_scons._release_tools.os.makedirs')
-    @mock.patch('gslab_scons._release_tools.os.path.isdir')
-    @mock.patch('gslab_scons._release_tools.shutil.rmtree')
-    @mock.patch('gslab_scons._release_tools.shutil.copy')
-    @mock.patch('gslab_scons._release_tools.shutil.make_archive')
-    @mock.patch('gslab_scons._release_tools.shutil.move')
-    def test_release_standard(self, mock_move, mock_make_archive, mock_copy, 
-                              mock_rmtree, mock_isdir, mock_makedirs, 
-                              mock_sleep, mock_getpass, mock_session,
-                              mock_upload):
+    @patch_collection
+    def test_release_standard(self, 
+                              mock_move, 
+                              mock_make_archive, 
+                              mock_copy, 
+                              mock_makedirs, 
+                              mock_getpass, 
+                              mock_session,
+                              mock_upload, 
+                              mock_rmtree,
+                              mock_isdir):
```

Comment:
All of the patches are referenced within `@patch_collection`, which I created to avoid having to include a long list of patches above each test method.